ocfweb/stats/job_frequency.py

```diff
@@ -0,0 +1,106 @@
+import urllib.parse
+from collections import defaultdict
+from datetime import date
+from datetime import datetime
+
+import numpy as np
+import ocflib.printing.quota as quota
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse
+from django.shortcuts import redirect
+from matplotlib.figure import Figure
+from matplotlib.ticker import MaxNLocator
+
+from ocfweb.caching import periodic
+from ocfweb.component.graph import plot_to_image_bytes
+
+
+def pyday_to_sqlday(pyday):
+    """Converting weekday index from python to mysql."""
+    return (pyday + 1) % 7 + 1
+
+
+def get_jobs(sql='SELECT * FROM `public_jobs`'):
+    """Get the publicjobs database to python"""
+    conn = quota.get_connection(db='ocfprinting')
+    with conn.cursor() as curs:
+        curs.execute(sql)
+        return curs.fetchall()
+
+
+@periodic(60)
+def _jobs_graph_image(day=None):
+    if not day:
+        day = date.today()
+
+    return HttpResponse(
+        plot_to_image_bytes(get_jobs_plot(day), format='svg'),
+        content_type='image/svg+xml',
+    )
+
+
+def daily_jobs_image(request):
+    try:
+        day = datetime.strptime(request.GET.get('date', ''), '%Y-%m-%d').date()
+    except ValueError:
+        day = date.today()
+
+    # redirect to canonical url
+    if request.GET.get('date') != day.isoformat():
+        return redirect('{}?{}'.format(
+            reverse('daily_job_image'),
+            urllib.parse.urlencode({'date': day.isoformat()}),
+        ))
+
+    if day == date.today():
+        return _jobs_graph_image()
+    else:
+        return _jobs_graph_image(day=day)
+
+
+def get_jobs_plot(day):
+    """Return matplotlib plot showing the number i-page-job to the day."""
+
+    day_of_week = pyday_to_sqlday(day.weekday())
+    day_quota = quota.daily_quota(datetime.combine(day, datetime.min.time()))
```

Comment:
the parameter day is actually a "date" object, but quota.daily_quota takes in "datetime" object